docs/documentation/advanced/introduction.md

```diff
@@ -0,0 +1,12 @@
+---
+layout: standard
+title: Introduction
+---
+
+This section focus on providing specifics examples for complex case.
+
+:::info{title="Important"}
+You don't need, to understand this section, to use the library.
```

Comment:
I am not sure about this sentence. I think this section (as of now) describes a not so rare case, or maybe even a typical case, when you get a JSON with a structure where keys and values are a map or dictionary. It is very common to have a structure like:

```json
{ "customers": { 
    { "Ramesh": { "age": 32, "address": "Ahmedbad" }
    , "Muffy": { "age": 44, "address": "Indore" } 
    }
}
```
instead of:
```json
{ "customers": 
    [ { "name": "Ramesh", "age": 32, "address": "Ahmedbad" }
    , { "name": "Muffy", "age": 44, "address": "Indore" } 
    ]
}
```

Why is the former covered in an advanced section with explanation that this is not really required to know and the latter is supposed to be super easy to decode, covered in basic, beginner friendly section?

Reply:
Well, in the example you gave just above all the properties are related to user and this is what is supported out of the box by Thoth.Json.

In your currency example, you have some fields of a `Currency` type and others field like `ts` that are to be ignored when getting the currency but is still needed in the end for the final type. For me, this is a first time seeing that kind of JSON and it make complex.

Also, most people are already fearful of the manual API because they need to write a small bit of code. So, if I was to show the kind of code there is inside this PR or in the original issue in the standard documentation, most of them would be even more scared of it. 

This is why, I am placing it inside of an advanced section which explicitly state that this is not required to understand that section to use Thoth.Json in general.